src/game/player_callbacks_1.c

```diff
@@ -1337,4 +1292,2097 @@ void PlayerCB_8027114(Player *player)
+        player->unk64 = 9;
+    }
+
+    sub_8023128(player);
+}
+END_NONMATCH
+
+void sub_802989C(Player *player)
+{
+    if ((u16)gRingCount >= 150) {
+        player->unk52 = 4;
+    } else if ((u16)gRingCount >= 100) {
+        player->unk52 = 3;
+    } else if ((u16)gRingCount >= 50) {
+        player->unk52 = 2;
+    } else if ((u16)gRingCount > 10) {
+        player->unk52 = 1;
+    } else {
+        player->unk52 = 0;
+    }
+}
+
+void sub_80298DC(Player *player)
+{
+    // TODO: unk5A Might be isBoostActive ?
+    bool32 isBoostActive = player->unk5A;
+    if (isBoostActive) {
+        if (!(player->moveState & MOVESTATE_IN_AIR)) {
+            player->unk58 = gUnknown_080D6916[player->unk52];
+
+            if (ABS(player->speedGroundX) < Q_24_8(4.5)) {
+                player->unk5A = FALSE;
+                player->unk58 = 0;
+            }
+        }
+    } else {
+        if ((!(player->moveState & MOVESTATE_IN_AIR))
+            && ((ABS(player->speedGroundX) >= player->unk44))) {
+            if (player->unk58 >= gUnknown_080D6916[player->unk52]) {
+                player->unk5A = TRUE;
+                gCamera.unk8 = 0x400;
+
+                CreateBoostModeParticles();
+
+                m4aSongNumStart(SE_221);
+            }
+        } else {
+            player->unk58 = isBoostActive;
+        }
+    }
+}
+
+void sub_8029990(Player *player)
+{
+    s32 absSpeed = ABS(player->speedGroundX);
+
+    if (absSpeed <= Q_24_8(1.25)) {
+        player->unk54 = 0;
+    } else if (absSpeed <= Q_24_8(2.5)) {
+        player->unk54 = 1;
+    } else if (absSpeed <= Q_24_8(4.0)) {
+        player->unk54 = 2;
+    } else if (absSpeed <= Q_24_8(9.0)) {
+        player->unk54 = 3;
+    } else if (absSpeed <= Q_24_8(10.0)) {
+        player->unk54 = 4;
+    } else {
+        player->unk54 = 5;
+    }
+}
+
+// NOTE: Proc type should be the same as sub_8021604!
+void sub_80299F0(u32 character, u32 level, u32 p2, Player *player)
+{
+    sub_8021604(character, level, p2, player);
+}
+
+void sub_80299FC(Player *player)
+{
+    TaskDestroy(player->spriteTask);
+    player->spriteTask = NULL;
+
+    if (player->unk60 == 0) {
+        sub_801F78C();
+        sub_8021350();
+    }
+}
+
+void *sub_8029A28(Player *player, u8 *p1, int *out)
+{
+    void *result;
+
+    u8 dummy;
+
+    // TODO: Why is dummyInt unused?
+    int dummyInt;
+    int p1Value;
+
+    if (p1 == NULL)
+        p1 = &dummy;
+    if (out == NULL)
+        out = &dummyInt;
+
+    result = sub_802195C(player, p1, out);
+
+    p1Value = *p1;
+
+    if (p1Value & 0x1)
+        *p1 = 0;
+    else {
+        if (gUnknown_03005424 & EXTRA_STATE__GRAVITY_INVERTED) {
+            s32 val = -0x80;
+            val -= p1Value;
+            *p1 = val;
+        }
+    }
+
+    return result;
+}
+
+void *sub_8029A74(Player *player, u8 *p1, int *out)
+{
+    void *result;
+
+    u8 dummy;
+
+    // TODO: Why is dummyInt unused?
+    int dummyInt;
```

Comment:
Minor but these shouldn't be int as I don't think we are supposed to use in GBA projects. Just a note for future

Reply:
I used an int because I do not know what the type actually is.
Though I guess I can replace it with s32?